pandas/core/groupby.py

```diff
@@ -2315,6 +2315,9 @@ def __init__(self, index, grouper=None, obj=None, name=None, level=None,
                 # groupby
                 else:
                     cat = self.grouper.unique()
+                    cat.add_categories([c for c in self.grouper.categories
+                                        if c not in cat.categories],
```

Comment:
How strongly would `self.grouper.categories[~self.grouper.categories.isin(cat.categories)]` be preferred?